### ISSUE TITLE
[v6] componentDidAppear not triggered when changing bottomtabs using mergeOptions

### ISSUE THREAD

**Issue by didrikkvanvik** [wix/react-native-navigation#6143]
### Issue Description

`componentDidAppear` is not triggered when changing the current `bottomTab` using `mergeOptions`. Everything works fine when pressing a bottomTab. I've used the recommended way of navigating between tabs, as stated in [the documentation](https://wix.github.io/react-native-navigation/docs/bottomTabs#changing-bottomtabs-visibility). I believe this is a result of having the 
`tabsAttachMode` in bottomTabs options set to `onSwitchToTab`

### Steps to Reproduce / Code Snippets / Screenshots
The following code is how i navigate to another tab:

```
Navigation.mergeOptions(componentId, { bottomTabs: { currentTabIndex: tabIndex } })
```
The `tabIndex` is the index of the tab we're switching to. 

---
### Environment
* React Native Navigation version: v6.0.1
* React Native version: v0.61.4
* Platform(s) (iOS, Android, or both?): iOS
* Device info (Simulator/Device? OS version? Debug/Release?): Simulator (Debug)

**Comment by guyca** [comment]
Hey @didrikkvanvik Have you tried updating to latest version? Also on which iOS version have you tested?
Are you sure this happens on both platforms?

**Comment by didrikkvanvik** [comment]
I've only tested on iOS so far with version 13.4.1. We're experiences another crucial [bug](https://github.com/wix/react-native-navigation/issues/6000) with react-native-navigation that prohibits us from updating to the latest version. After patching the bug, we tried to update to the latest RNN version, but the bug still occurs even with the patch. So I haven't tried with the latest RNN version. Can't seem to find anything in the changelog that would fix this issue though since v6.0.1 (correct me if i'm wrong).
